src/plugins/string/StringFix.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * StringFix.js
+ *
+ * Copyright © 2023 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Fix } from "i18nlint-common";
+import { StringFixCommand } from "./StringFixCommand.js";
+
+export class StringFix extends Fix {
+    /**
+     * @override
+     * Matches IRs produced by {@link StringParser}
+     */
+    type = "string";
+
+    /**
+     * @readonly
+     * @type {StringFixCommand[]}
+     */
+    commands = [];
+
+    constructor(/** @type {StringFixCommand[]} */ ...commands) {
```

Comment:
I disagree. The idea here is that the Fix is a set of instructions that tell the Fixer how to remediate an issue (a single Fix for a single Result). Because of that, to me it makes sense for the instruction set to be immutable - upon creation of a Fix you should already know all steps necessary for fixing; the Fix is not concerned about how You get them.

By exposing a `Fix.addCommand()` you basically tell whomever's using this fix that it is OK to add to it at will - which is not the case, since the original intention is that a Fix should only be created by a `Rule.match()`.

About Your example - are we not producing a separate Result for each match of a regexp rule? It makes sense to me that given a string `テスト。 テスト。 テスト。` for the following results
1. there should be no space after japanese fullstop: `テスト>>。 <<テスト。 テスト。`
2. there should be no space after japanese fullstop: `テスト。 テスト>>。 <<テスト。`
each of them would have its own fix created like
```js
new Result({ /* ... */, fix: new StringFix(StringFixCommand.deleteAfter(match.index, match[0].length)) })
```

Had the rule been written in such a way that there would be only one Result with multiple highlights throughout the whole string, and had it been written in such a way that there is a while or for-of loop over all the matches, You would need to accumulate these partial results outside of the loop anyway - so it also makes sense to do the same thing with each partial fix for each partial result.

As for the rest params, we can get rid of it and just require an array - it was mostly there as a sugar to get 
```js
new StringFix(StringFixCommand.deleteAfter(match.index, match[0].length))
```
over
```js
new StringFix([StringFixCommand.deleteAfter(match.index, match[0].length)])
```
in one-command fixes (because conversely, I don't like the approach `commands: StringFixCommand | StringFixCommand[]` and then doing `if (!Array.isArray(commands)) { commands = [commands]; }`).

Reply:
I don't feel strongly enough about this one to argue more about it, so it if works as it is, it's fine.